src/multivariate/jointorderstatistics.jl

```diff
@@ -0,0 +1,132 @@
+# Implementation based on chapters 2-4 of
+# Arnold, Barry C., Narayanaswamy Balakrishnan, and Haikady Navada Nagaraja.
+# A first course in order statistics. Society for Industrial and Applied Mathematics, 2008.
+
+"""
+    JointOrderStatistics <: ContinuousMultivariateDistribution
+
+The joint distribution of a subset of order statistics from a sample from a continuous univariate distribution.
+
+    JointOrderStatistics(
+        dist::ContinuousUnivariateDistribution,
+        n::Int,
+        r::AbstractVector{Int}=1:n;
+        check_args::Bool=true,
+    )
+
+Construct the joint distribution of order statistics `r` from an IID sample of size `n` from `dist`.
+
+The ``i``th order statistic of a sample is the ``i``th element of the sorted sample.
+For example, the 1st order statistic is the sample minimum, while the ``n``th order
+statistic is the sample maximum.
+
+`r` must be a sorted vector of unique integers between 1 and `n`.
+
+For a single order statistic, use [`OrderStatistic`](@ref) instead.
+
+## Examples
+
+```julia
+JointOrderStatistics(Normal(), 10)           # Product(fill(Normal(), 10)) restricted to ordered vectors
+JointOrderStatistics(Cauchy(), 10, [1, 10])  # joint distribution of the extrema
+```
+"""
+struct JointOrderStatistics{D<:ContinuousUnivariateDistribution,R<:AbstractVector{Int}} <:
+       ContinuousMultivariateDistribution
+    dist::D
+    n::Int
+    r::R
+    function JointOrderStatistics(
+        dist::ContinuousUnivariateDistribution,
+        n::Int,
+        r::AbstractVector{Int}=1:n;
+        check_args::Bool=true,
+    )
+        @check_args(
+            JointOrderStatistics,
+            (n, n ≥ 1, "`n` must be a positive integer."),
+            (
+                r,
+                1 ≤ first(r) && last(r) ≤ n && issorted(r) && allunique(r),
```

Comment:
This could be checked more efficiently by just iterating through `r` once.

Reply:
Good point, and it turns out that `issorted` can handle this if we just provide a function to `lt` that is also adds an equality check; this is much more efficient than calling `allunique`.